src/content/vrf/v2-5/migration-from-v1.mdx

```diff
@@ -0,0 +1,306 @@
+---
+section: vrf
+date: Last Modified
+title: "Migrating from VRF v1"
+---
+
+import Vrf2_5Common from "@features/vrf/v2-5/Vrf2_5Common.astro"
+import { Aside, CodeSample } from "@components"
+import { Tabs, TabsContent } from "@components/Tabs"
+
+<Vrf2_5Common callout="security" />
+
+VRF V2.5 replaces both VRF V1 and VRF V2 on November 29, 2024. [Learn more about VRF V2.5](https://blog.chain.link/introducing-vrf-v2-5/).
+
+## Comparing VRF v1 to VRF v2.5
+
+Chainlink VRF v2.5 includes several improvements and changes to the way you fund and request randomness for your smart contracts:
+
+- You have the option to manage payment for your VRF requests by pre-funding a subscription account, or to directly fund your consuming contracts as you do with VRF V1. [Compare subscription and direct funding](/vrf/#choosing-the-correct-method).
+- VRF v2.5 introduces the option to pay for requests in either LINK or native tokens. This choice is available for both subscription and direct funding.
+
+### New billing options
+
+- **Native billing:** You have the option to use either native tokens or LINK to pay for VRF requests. Instead of a flat LINK fee per request, there is percentage-based premium fee applied to each request. Refer to the [Billing](/vrf/v2-5/billing) page for more details. To find out the premium percentages for the networks you use, see the [Supported Networks](/vrf/v2-5/supported-networks) page.
```

Comment:
```suggestion
- **Native billing:** You have the option to use either native tokens or LINK to pay for VRF requests. Instead of a flat LINK fee per request, there is percentage-based premium fee applied to each request. See the [Billing](/vrf/v2-5/billing) page for more details. To find out the premium percentages for the networks you use, see the [Supported Networks](/vrf/v2-5/supported-networks) page.
```